test/functional/test_runner.py

```diff
@@ -728,47 +729,59 @@ def get_next(self):
             test_argv = test.split()
             testdir = "{}/{}_{}".format(self.tmpdir, re.sub(".py$", "", test_argv[0]), portseed)
             tmpdir_arg = ["--tmpdir={}".format(testdir)]
-            self.jobs.append((test,
-                              time.time(),
-                              subprocess.Popen([sys.executable, self.tests_dir + test_argv[0]] + test_argv[1:] + self.flags + portseed_arg + tmpdir_arg,
-                                               text=True,
-                                               stdout=log_stdout,
-                                               stderr=log_stderr),
-                              testdir,
-                              log_stdout,
-                              log_stderr))
+
+            def proc_wait(task):
+                task[2].wait()
+                return task
+
+            task = [
+                test,
+                time.time(),
+                subprocess.Popen(
+                    [sys.executable, self.tests_dir + test_argv[0]] + test_argv[1:] + self.flags + portseed_arg + tmpdir_arg,
+                    text=True,
+                    stdout=log_stdout,
+                    stderr=log_stderr,
+                ),
+                testdir,
+                log_stdout,
+                log_stderr,
+            ]
+            fut = self.executor.submit(proc_wait, task)
+            self.jobs[fut] = test
         if not self.jobs:
             raise IndexError('pop from empty list')
```

Comment:
My understanding is that in Python a list `self.test_list.pop(0)` is O(n) so consuming all elements is quadratic this way.
A `collections.deque` with `popleft` might describe the desired behavior better (performance-wise it would likely be the same for a few hundred elements)

nit: `jobs` isn't a "list" anymore

Reply:
> so you should be able to create a separate pull

yes, I think it should be done in a separate PR, should have clarified that.

> Yes, the message is "wrong".

I'm not blocking, I don't think it's important, but if you edit again, maybe we can remove/update the message to remove the confusion